### PR TITLE
Transfer Component - Refresh

### PR DESCRIPTION
**Pull request by nalancer08**
### Type of change

- [ ] Chore
- [X] Feature
- [ ] Bug fix

### Why

Transfer Component UI with new updates, includes:

- CDecimal class to handle bin numbers with deciamals
- AssetFormatter class to allow transform big numbers into currency numbers: 1000 --> 10.00 USD
- New UI for Transfer Component with last design review
- Fix problems with fiat global currency to create transfers (deposit, withdraw)
- New Test for ViewModels

#### Modal Error

https://user-images.githubusercontent.com/7268597/229025667-ba32a96f-40e3-4b31-ac5a-37357cbed65e.mov

### CODE REVIEW

File: CybridSDK/Core/CDecimal.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+            if decimalPart.count < 2 {
+                self.decimalValue = "\(String(decimalPart))0"
+            } else {
+                self.decimalValue = String(decimalPart)
+            }
+        } else {
+
+            self.intValue = BigInt(value) ?? BigInt(0)
+            self.decimalValue = String(repeating: "0", count: 2)
```
Comment (brent-cybrid) [inline CybridSDK/Core/CDecimal.swift:48]: Why are we using '2' here and on line 40. We should be taking as input the asset and using the number of decimal places based on the asset.

File: CybridSDK/Framework Extensions/UIView/String+Extensions.swift [hunk 2]
```diff
@@ -14,8 +14,18 @@ extension String {
...
+        // Check if has dot
+        if self.contains(".") {
+
+            let parts = self.stringValue.split(separator: ".")
+            intValue = String(parts[0])
+            decimalValue = String(parts[1])
+        }
+
+        for character in intValue.reversed() {
             if iterator < 4 {
```
Comment (brent-cybrid) [inline CybridSDK/Framework Extensions/UIView/String+Extensions.swift:29]: Why do we have 4 here?

File: CybridSDK/Utilities/Formatters/AssetFormatter.swift [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+    /// Method to set amount in any asset in format to trade (the amount always handle in input)
+    /// Example:
+    /// - 10 USD --> 1000
+    /// - 10.5 USD --> 1050
+    /// - 0.064218090000000000 ETH --> 64218090000000000
+    /// Parameters:
+    ///  - asset: AssetBankModel to take the decimals and symbol ($)
+    ///  - amount: CDecimal object with preloaded amount
+    static func forTrade(_ asset: AssetBankModel, amount: CDecimal) -> String {
```
Comment (brent-cybrid) [inline CybridSDK/Utilities/Formatters/AssetFormatter.swift]: We should choose a better name for this method as we'll use it in places other than just trade. How about 'fromInput?' or something like that.

File: CybridSDKTests/Core/CDecimalTest.swift [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        let decimal = CDecimal(valueToConvert)
+
+        // -- Then
+        XCTAssertEqual(decimal.originalValue, valueToConvert)
+        XCTAssertEqual(decimal.intValue, decimalIntValue)
+        XCTAssertEqual(decimal.decimalValue, decimalDecimalValue)
+        XCTAssertEqual(decimal.newValue, decimalnewValue)
+    }
+
+    func test_init_String_Int_Error_With_Decimal() {
```
Comment (brent-cybrid) [inline CybridSDKTests/Core/CDecimalTest.swift:86]: Shouldn't these test cases raise an exception?

**Review by brent-cybrid — APPROVED** [review]

File: CybridSDK/Core/CDecimal.swift (hunk 1 above)
Comment (nalancer08) [reply CybridSDK/Core/CDecimal.swift:48]: The idea here is that this class don't need `AssetBankModel` for that reason is always at 2 decimals, then the AssetFormatter checks if the decimals needs more zeros and that class add the zeros.

File: CybridSDK/Utilities/Formatters/AssetFormatter.swift (hunk 3 above)
Comment (nalancer08) [reply CybridSDK/Utilities/Formatters/AssetFormatter.swift]: Perfect! Renamed!

File: CybridSDKTests/Core/CDecimalTest.swift (hunk 4 above)
Comment (nalancer08) [reply CybridSDKTests/Core/CDecimalTest.swift:86]: The idea of this class is not return null objects to don't have to handle nullable verification in each transaction that add a lot of boilerplate code. So the class check if it's not possible construct de BigInt it returns zero, so in UI I don't hav to check this:

- Empty numbers to cast to 0
- NaN numbers to be equal to 0

Just set the number and if its zero all is zero and into logger we show the problem

File: CybridSDK/Framework Extensions/UIView/String+Extensions.swift (hunk 2 above)
Comment (nalancer08) [reply CybridSDK/Framework Extensions/UIView/String+Extensions.swift:29]: For some reason iOS don't have formatter for String numbers, just for NSNumbers, so this class allow me to set a number in String with format:

1000 --> 1,000

So the idea of this for is check if the integer value has more than 3 digits to have sense part the String and set the separator.

**Pull request merged**